src/macros.rs

```diff
@@ -163,11 +163,11 @@ macro_rules! impl_fixed_array_conversions {
             }
         }
 
-        impl<$S: Clone> From<[$S; $n]> for $ArrayN<$S> {
+        impl<$S> From<[$S; $n]> for $ArrayN<$S> {
             #[inline]
             fn from(v: [$S; $n]) -> $ArrayN<$S> {
                 // We need to use a clone here because we can't pattern match on arrays yet
```

Comment:
Is this comment stale now? 🤔

Reply:
Maybe we could just remove it. Good stuff! 👍

Reply:
Wait.... is this really safe though? What happens if we pass something with a destructer in? Or custom clone impl? Exceedingly unlikely, but we might be violating the invariants of the type system here...

Reply:
If $S has a Drop impl wouldn't it be called when the resulting array is dropped?  We still only have one copy of the internal data.  BaseNum also requires Copy so custom clone is out of the question.

Reply:
Oh wait I need a mem::forget for the custom Drops.

Reply:
Isn't the necessity of that also removed by the Copy restraint though come to think of it?

Reply:
`Copy` types can still impl `Drop`. We probably would want to protect this impl anyway with some bounds in this case. With `T: Copy` at least. Again, this is a very unlikely case due to the fact that this will only likely ever be taking floats or ints, but it's important to be wary around `unsafe` and be good citizens of the ecosystem!

Reply:
If a copy type impls drop it's inherently unsafe as the drop call might be called multiple times and treated as the same data.

Reply:
I've added the restraint on the impl anyways